tests/raft_scenarios_runner.py

```diff
@@ -1,91 +1,100 @@
 def block(fd, title, level, lang=None, lines=None):
-    fd.write(level * "#" + title + "\n\n")
+    fd.write(level * "#" + " " + title + "\n\n")
     fd.write("```" + (lang or "") + "\n")
     for line in lines or []:
         fd.write(line + "\n")
     yield
     fd.write("\n```\n\n")
 
 
-@contextmanager
-def error_report_block(fd, errors=None):
-    fd.seek(0, 0)
+def write_error_report(errors=None):
     if errors:
-        fd.write("???+ error \n")
-        fd.write(4 * " " + "| Scenario | stderr |\n")
-        fd.write(4 * " " + "|----------|--------|\n")
+        errors = [(error[0], error[1].replace("\n", " <br> ")) for error in errors]
+        scenario_len = max(len("Scenario"), *(len(error[0]) for error in errors))
+        stderr_len = max(len("stderr"), *(len(error[1]) for error in errors))
+        print("???+ error \n")
+        fmt_s = "   | {{:<{}}} | {{:<{}}} |\n".format(scenario_len, stderr_len)
+        print(fmt_s.format("Scenario", "stderr"))
+        print(fmt_s.format("-" * scenario_len, "-" * stderr_len))
         for error in errors:
-            fd.write(4 * " " + "| " + error[0] + " | ")
-            stderr_lines = error[1].split("\n")
-            for err in stderr_lines:
-                if err:
-                    fd.write(4 * " " + "```" + err + "``` <br>")
-            fd.write(4 * " " + "|\n")
+            print(fmt_s.format(error[0], error[1]))
     else:
-        fd.write("??? success \n")
-    yield
-
-
-def prepend_error_report(fd, errors=None):
-    content = raft.read()
-    raft.seek(0, 0)
-    with error_report_block(raft, errors):
-        raft.write("***\n")
-    raft.write(content)
+        print("??? success \n")
 
 
 def strip_log_lines(text):
     ol = []
     for line in text.split(os.linesep):
-        if not line.startswith("["):
-            ol.append(line)
+        if line.startswith("<RaftDriver>"):
+            ol.append(line[len("<RaftDriver>") :])
     return os.linesep.join(ol)
 
 
+def expand_files(files):
+    all_files = []
+    for path in files:
+        if os.path.isdir(path):
+            for dirpath, _, filenames in os.walk(path):
+                for name in filenames:
+                    all_files.append(os.path.join(dirpath, name))
+        else:
+            all_files.append(path)
+    return all_files
+
+
 if __name__ == "__main__":
-    driver, path, doc = sys.argv[1], sys.argv[2], sys.argv[3]
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("driver", type=str, help="Path to raft_driver binary")
+    parser.add_argument("--gen-scenarios", action="store_true")
+    parser.add_argument("files", nargs="+", type=str, help="Path to scenario files")
+
+    args = parser.parse_args()
+
     err_list = []
     test_result = True
 
-    generate_scenarios(path)
+    files = expand_files(args.files)
+
+    if args.gen_scenarios:
+        files += generate_scenarios()
+
+    ostream = sys.stdout
 
-    with open(os.path.join(doc, RAFT_TEST_FILE_NAME), "w") as raft:
```

Comment:
Making this test more like all our other test apps. Rather than writing to a file, it writes to stdout. It still strips out the mermaid-specific bits, and surrounds them with some Markdown decoration, but it dumps any errors at the end rather than the start (for access from CTest). Can then be piped to a `.md` file to preview the resulting trace diagrams.